src/components/tl/sharp/tl_sharp.c

```diff
@@ -30,7 +30,7 @@ static ucc_config_field_t ucc_tl_sharp_context_config_table[] = {
     {"", "", NULL, ucc_offsetof(ucc_tl_sharp_context_config_t, super),
      UCC_CONFIG_TYPE_TABLE(ucc_tl_context_config_table)},
 
-    {"DEVICES", "mlx5_0:1",
+    {"DEVICES", "",
```

Comment:
how device will be selected for previous SHARP releases?

Reply:
right, this is my question, if one builds UCC 1.2 with SHARP 3.0 how it will work without providing device?

Reply:
the default selection was fixed in libsharp in the last HPCX release  i. it will fail if we use older than that. that should be ok